packages/hardhat-core/src/internal/core/config/config-env.ts

```diff
@@ -173,3 +177,25 @@ export function experimentalAddHardhatNetworkMessageTraceHook(
   const ctx = HardhatContext.getHardhatContext();
   ctx.experimentalHardhatNetworkMessageTraceHooks.push(hook);
 }
+
+/**
+ * Secrets manager functions
+ */
+export const secrets = {
+  get: getSecret,
+};
+
+function getSecret(key: string, defaultValue?: string): string {
+  const secretsManager = new SecretsManager(getSecretsFilePath());
+
+  const value = secretsManager.get(key);
+
+  if (value !== undefined) return value;
+
+  if (defaultValue !== undefined) return defaultValue;
+
+  throw new HardhatError(ERRORS.ARGUMENTS.VALUE_NOT_FOUND_FOR_ARGUMENT, {
```

Comment:
I think this error is too generic and will be a bit confusing.

Maybe we should create a new error category for secrets (like we have now for arguments, artifacts, plugins, etc.)? Since we already added three, and I can imagine us adding more.